.gitlab/ci/.gitlab-ci.sdk-nightly.yml

```diff
@@ -264,21 +320,187 @@ demisto-sdk-nightly:run-end-to-end-tests:
     ARTIFACTS_FOLDER_INSTANCE: "${ARTIFACTS_FOLDER}/instance_${INSTANCE_ROLE}"
     ARTIFACTS_FOLDER_SERVER_TYPE: "${ARTIFACTS_FOLDER}/server_type_${SERVER_TYPE}"
     ENV_RESULTS_PATH: "${ARTIFACTS_FOLDER_SERVER_TYPE}/env_results.json"
+    FAIL_ON_ERROR: "true"
+    E2E_TEST_FOLDER: "xsoar"
   needs: ["demisto-sdk-nightly:xsoar-prepare-testing-bucket"]
   stage: run-instances
   script:
+    - !reference [.ssh-config-setup]
+
+    - !reference [.copy_env_results]
+
+    - section_start "Wait Until Server Ready" --collapsed
+    - |
+      [ -n "${NIGHTLY}" ] && IS_NIGHTLY=true || IS_NIGHTLY=false
+      python3 ./Tests/scripts/wait_until_server_ready.py  -n ${IS_NIGHTLY} --instance-role "${INSTANCE_ROLE}" || EXIT_CODE=$?
+    - section_end "Wait Until Server Ready"
+
+    - !reference [.download-demisto-conf]
+    - !reference [.secrets-fetch]
+
+    - section_start "End to End Tests"  --collapsed
     - |
       mkdir $ARTIFACTS_FOLDER/demisto-sdk
       git clone -b ${SDK_REF} --single-branch --depth 1 https://github.com/demisto/demisto-sdk.git $ARTIFACTS_FOLDER/demisto-sdk
 
-      TUNNEL_PORT=$(cat "${ENV_RESULTS_PATH}" | jq "map(select(.Role == \"$INSTANCE_ROLE\")) | .[].TunnelPort")
+      unset DEMISTO_API_KEY # a temporary fix
```

Comment:
why it's a temporary fix?

Reply:
after talking to Tal, it's actually not temporary, will remove the comment